### PR TITLE
seq

### PR DESCRIPTION
**Pull request by lilactown**
Please answer the following questions and leave the below in as part of your PR.

- [x] This PR corresponds to an [issue with a clear problem statement](https://github.com/babashka/babashka/blob/master/doc/dev.md#start-with-an-issue-before-writing-code).
- [x] This PR provides tests for code additions and changes

---

Related #22

This PR implements the following:
- Renames `iterable?` to `seqable?` and adds support for `nil` and `js/undefined`
- Updates `iterable` to return an empty array when `nil` or `undefined` are passed
- Adds a new `seq` function which is just like `iterable` but returns `nil` when the  iterable is empty.
- Updates `first`, `second` and `rest` to call `iterable` before destructuring.
- Tests for `seq`

This now allows us to do things like the following:

``` clojure
(loop [evens []
       nums (range 10)]
  (if-some [x (first nums)]
    ,,,))
```

### CODE REVIEW

File: core.js [hunk 1]
```diff
@@ -1,8 +1,6 @@
 export function assoc_BANG_(m, k, v, ...kvs) {
   if (kvs.length % 2 !== 0) {
-    throw new Error(
-      "Illegal argument: assoc expects an odd number of arguments."
-    );
+    throw new Error('Illegal argument: assoc expects an odd number of arguments.');
```
Comment (lilactown) [inline core.js:3]: some of these changes come from prettier reformatting. sorry for the noise. I hope it matches the style intended

Comment (borkdude) [reply core.js:3]: No problem

File: core.js [hunk 2]
```diff
@@ -217,31 +215,44 @@ export function get(coll, key, otherwise = undefined) {
   return key in coll ? coll[key] : otherwise;
 }
 
-export function iterable_QMARK_(x) {
+export function seqable_QMARK_(x) {
   // String is iterable but doesn't allow `m in s`
-  return x instanceof String || Symbol.iterator in x;
+  return typeof x === 'string' || x === null || x === undefined || Symbol.iterator in x;
```
Comment (lilactown) [inline core.js:220]: apparently this was totally broken for strings before

**Comment by lilactown** [comment]
I think this is ready for review

**Pull request merged**